x-pack/plugins/security_solution/public/common/containers/sourcerer/index.tsx

```diff
@@ -0,0 +1,415 @@
+ */
+
+import { get, noop, isEmpty } from 'lodash/fp';
+import React, { createContext, useCallback, useContext, useEffect, useReducer } from 'react';
+import { IIndexPattern } from 'src/plugins/data/public';
+
+import { NO_ALERT_INDEX } from '../../../../common/constants';
+import { useKibana } from '../../lib/kibana';
+
+import { SourceQuery } from '../../../graphql/types';
+
+import { sourceQuery } from '../source/index.gql_query';
+import { useApolloClient } from '../../utils/apollo_context';
+import {
+  sourceGroups,
+  SOURCE_GROUPS,
+  SourceGroupsType,
+  SOURCERER_FEATURE_FLAG_ON,
+} from './constants';
+import {
+  BrowserFields,
+  DocValueFields,
+  EMPTY_BROWSER_FIELDS,
+  EMPTY_DOCVALUE_FIELD,
+  getBrowserFields,
+  getDocValueFields,
+  getIndexFields,
+  indicesExistOrDataTemporarilyUnavailable,
+} from './format';
+
+// TYPES
+interface ManageSource {
+  browserFields: BrowserFields;
+  defaultPatterns: string[];
+  docValueFields: DocValueFields[];
+  errorMessage: string | null;
+  id: SourceGroupsType;
+  indexPattern: IIndexPattern;
+  indexPatterns: string[];
+  indicesExist: boolean | undefined | null;
+  loading: boolean;
+}
+
+interface ManageSourceInit extends Partial<ManageSource> {
+  id: SourceGroupsType;
+}
+
+type ManageSourceGroupById = {
+  [id in SourceGroupsType]?: ManageSource;
+};
+
+type ActionManageSource =
+  | {
+      type: 'SET_SOURCE';
+      id: SourceGroupsType;
+      defaultIndex: string[];
+      payload: ManageSourceInit;
+    }
+  | {
+      type: 'SET_IS_SOURCE_LOADING';
+      id: SourceGroupsType;
+      payload: boolean;
+    }
+  | {
+      type: 'SET_ACTIVE_SOURCE_GROUP_ID';
+      payload: SourceGroupsType;
+    }
+  | {
+      type: 'SET_AVAILABLE_INDEX_PATTERNS';
+      payload: string[];
+    }
+  | {
+      type: 'SET_IS_INDEX_PATTERNS_LOADING';
+      payload: boolean;
+    };
+
+interface ManageSourcerer {
+  activeSourceGroupId: SourceGroupsType;
+  availableIndexPatterns: string[];
+  availableSourceGroupIds: SourceGroupsType[];
+  isIndexPatternsLoading: boolean;
+  sourceGroups: ManageSourceGroupById;
+}
+
+export interface UseSourceManager extends ManageSourcerer {
+  getManageSourceGroupById: (id: SourceGroupsType) => ManageSource;
+  initializeSourceGroup: (
+    id: SourceGroupsType,
+    indexToAdd?: string[] | null,
+    onlyCheckIndexToAdd?: boolean
+  ) => void;
+  setActiveSourceGroupId: (id: SourceGroupsType) => void;
+  updateSourceGroupIndicies: (id: SourceGroupsType, updatedIndicies: string[]) => void;
+}
+
+// DEFAULTS/INIT
+export const getSourceDefaults = (id: SourceGroupsType, defaultIndex: string[]) => ({
+  browserFields: EMPTY_BROWSER_FIELDS,
+  defaultPatterns: defaultIndex,
+  docValueFields: EMPTY_DOCVALUE_FIELD,
+  errorMessage: null,
+  id,
+  indexPattern: getIndexFields(defaultIndex.join(), []),
+  indexPatterns: defaultIndex,
+  indicesExist: indicesExistOrDataTemporarilyUnavailable(undefined),
```

Comment:
Could we keep it as undefined or null as default value instead of converting into true?

Reply:
As some components may be mislead by this, start fetching data, and cause some errors on the client side.

Reply:
I'm not sure why this was like this before (in useWithSource), but I imagine it solved a bug somewhere. Afraid to create it again

Reply:
yes that important to keep as null at the beginning.

Reply:
@XavierM im confused if you're saying to change it or not??